src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -869,6 +868,22 @@ private SafeCurlSlistHandle SetRequestHeaders(SafeCurlHandle handle, HttpRequest
             return retVal;
         }
 
+        /// <summary>
+        /// Add request headers to curl API
+        /// </summary>
+        /// <param name="headers"></param>
+        /// <param name="handle"></param>
+        /// <param name="rawHandle"></param>
+        private static void AddRequestHeaders(HttpHeaders headers, SafeCurlSlistHandle handle, ref IntPtr rawHandle)     
+        {          
+            foreach (KeyValuePair<string, IEnumerable<string>> header in headers)
+            {              
+                string headerStr = header.Key + ": " +  headers.GetHeaderString(header.Key);                       
+                rawHandle = Interop.libcurl.curl_slist_append(rawHandle, headerStr);
+                handle.SetHandle(rawHandle);
+            }
```

Comment:
Also, I just looked at the docs for curl_slist_append, which state:
"A null pointer is returned if anything went wrong, otherwise the new list pointer is returned."
So here and anywhere else we're using it we should be validating the result.